Adds documentation for new products & purchase order endpoints

rickharrison opened the pull request.

rickharrison committed:
Adds documentation for POST /products

rickharrison committed:
Adds PUT /products/{productId}

rickharrison committed:
Centralizes POST/PUT products with component schema

rickharrison committed:
Adds list categories and sub-categories

rickharrison committed:
Adds example errors for products.

rickharrison committed:
Adds PUT /products/{productId}/options

rickharrison committed:
Adds POST /purchase-orders

rickharrison committed:
Adds PUT /purchase-orders/${purchaseOrderId}

rickharrison committed:
Adds purchase order errors

rickharrison committed:
Updates changelog with new endpoints.

jwickens commented:
Don't see

 ```   
   thcAmount,
    thcAmountEach,
    thcAmountRangeLow,
    thcAmountRangeHigh,
    thcUnit,
    cbdAmount,
    cbdAmountEach,
    cbdAmountRangeLow,
    cbdAmountRangeHigh,
    cbdUnit
```
    
Not sure if we just don't want to document for now or it was missed with the latest changes.

rickharrison committed:
Adds some more information to the inventory guide.

rickharrison committed:
Adds THC/CBD properties to create/update product

rickharrison commented:
Just added those!

jwickens committed:
Updates to PUT/POST product compounds.
- Update description for units
- Add compounds to return schema
- Add nullable true to these properties.
